server/src/main/java/org/apache/cassandra/sidecar/livemigration/LiveMigrationInstanceMetadataUtil.java

```diff
@@ -0,0 +1,193 @@
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.cassandra.sidecar.cluster.instance.InstanceMetadata;
+import org.jetbrains.annotations.NotNull;
+
+import static org.apache.cassandra.sidecar.common.ApiEndpointsV1.LIVE_MIGRATION_CDC_RAW_DIR_PATH;
+import static org.apache.cassandra.sidecar.common.ApiEndpointsV1.LIVE_MIGRATION_COMMITLOG_DIR_PATH;
+import static org.apache.cassandra.sidecar.common.ApiEndpointsV1.LIVE_MIGRATION_DATA_FILE_DIR_PATH;
+import static org.apache.cassandra.sidecar.common.ApiEndpointsV1.LIVE_MIGRATION_HINTS_DIR_PATH;
+import static org.apache.cassandra.sidecar.common.ApiEndpointsV1.LIVE_MIGRATION_LOCAL_SYSTEM_DATA_FILE_DIR_PATH;
+import static org.apache.cassandra.sidecar.common.ApiEndpointsV1.LIVE_MIGRATION_SAVED_CACHES_DIR_PATH;
+import static org.apache.cassandra.sidecar.livemigration.LiveMigrationPlaceholderUtil.CDC_RAW_DIR_PLACEHOLDER;
+import static org.apache.cassandra.sidecar.livemigration.LiveMigrationPlaceholderUtil.COMMITLOG_DIR_PLACEHOLDER;
+import static org.apache.cassandra.sidecar.livemigration.LiveMigrationPlaceholderUtil.DATA_FILE_DIR_PLACEHOLDER;
+import static org.apache.cassandra.sidecar.livemigration.LiveMigrationPlaceholderUtil.HINTS_DIR_PLACEHOLDER;
+import static org.apache.cassandra.sidecar.livemigration.LiveMigrationPlaceholderUtil.LOCAL_SYSTEM_DATA_FILE_DIR_PLACEHOLDER;
+import static org.apache.cassandra.sidecar.livemigration.LiveMigrationPlaceholderUtil.SAVED_CACHES_DIR_PLACEHOLDER;
+
+/**
+ * Utility class for having all {@link InstanceMetadata} related helper functions related to
+ * Live Migration in one place.
+ */
+@SuppressWarnings("ConstantValue")
+public class LiveMigrationInstanceMetadataUtil
+{
+
+    /**
+     * List of directories Live Migration considers to transfer between source and destination.
+     */
+    public static List<String> dirsToCopy(InstanceMetadata instanceMetadata)
+    {
+        List<String> dirsToCopy = new ArrayList<>();
+        dirsToCopy.add(instanceMetadata.hintsDir());
+        dirsToCopy.add(instanceMetadata.commitlogDir());
+        if (instanceMetadata.savedCachesDir() != null)
+        {
+            dirsToCopy.add(instanceMetadata.savedCachesDir());
+        }
+        if (instanceMetadata.cdcDir() != null)
+        {
+            dirsToCopy.add(instanceMetadata.cdcDir());
+        }
+        if (instanceMetadata.localSystemDataFileDir() != null)
+        {
+            dirsToCopy.add(instanceMetadata.localSystemDataFileDir());
+        }
+        dirsToCopy.addAll(instanceMetadata.dataDirs());
+        return dirsToCopy;
+    }
+
+    /**
+     * Returns map of directory that can be copied and the index to use while constructing
+     * url for file transfer.
+     */
+    public static Map<String, String> dirPathPrefixMap(InstanceMetadata instanceMetadata)
+    {
+        Map<String, String> dirIndexMap = new Hashtable<>();
```

Comment:
Any reason why we would use a hashtable here instead of a HashMap?
```suggestion
        Map<String, String> dirIndexMap = new HashMap<>();
```

Reply:
HashMap allows null key and values. dirIndexMap should not have either null key or null value.

Reply:
in general use of Hashtable is discouraged. It's an old class and it is synchronized, the recommendation is the usage of HashMap. I don't see any null keys/values used in the scope of this method. Are you thinking the map will be updated outside of this method? And maybe if we don't expect changes to the map returned in this method we can return an unmodifiable map here?
```
return Collections.unmodifiableMap(dirIndexMap);
```

Reply:
Ran into null keys while unit testing (as some of the directories are optional). Changed it from HashMap to HashTable to catch null keys early. Null checks are already added to optional values. Changed it to HashMap and wrapped it with unmodifiable before returning.